solarforecastarbiter/metrics/tests/test_calculator.py

```diff
@@ -145,23 +184,76 @@ def test_calculate_metrics_single(single_forecast_data_obj,
+    results = calculator.calculate_metrics([proc_prfx_obs], LIST_OF_CATEGORIES,
+                                           PROB_NO_REF)
+    assert isinstance(results, list)
+    assert len(results) == 1
+    assert isinstance(results[0], tuple)
+    single_results, dist_results = results[0]
+    assert len(single_results) == len(const_values)
+    assert all(isinstance(r, datamodel.MetricResult) for r in single_results)
+    assert isinstance(dist_results, datamodel.MetricResult)
+
+    # With reference
+    ref_fx_values = pd.DataFrame(np.random.randn(10, 3)+10,
+                                 index=create_dt_index(10))
+    conv_ref_prob_fx = copy_prob_forecast_with_axis(
+        prob_forecasts, prob_forecasts.axis, constant_values=const_values)
+    ref_prfxobs = datamodel.ForecastObservation(conv_ref_prob_fx,
+                                                single_observation)
+    proc_ref_prfx_obs = create_processed_fxobs(ref_prfxobs,
+                                               ref_fx_values,
+                                               obs_values)
+    results = calculator.calculate_metrics([proc_prfx_obs], LIST_OF_CATEGORIES,
+                                           PROBABILISTIC_METRICS,
+                                           ref_pair=proc_ref_prfx_obs)
+    assert isinstance(results, list)
+    assert len(results) == 1
+    assert isinstance(results[0], tuple)
+    single_results, dist_results = results[0]
+    assert len(single_results) == len(const_values)
+    assert all(isinstance(r, datamodel.MetricResult) for r in single_results)
+    assert isinstance(dist_results, datamodel.MetricResult)
+
 
-    # Error - ProbabilisticForecast not yet supported
-    with pytest.raises(NotImplementedError):
-        calculator.calculate_metrics([proc_prfx_obs],
-                                     categories, metrics)
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+def test_calculate_metrics_with_probablistic_no_ref(single_observation,
+                                                    prob_forecasts,
+                                                    create_processed_fxobs,
+                                                    create_dt_index,
+                                                    copy_prob_forecast_with_axis):  # NOQA
+    const_values = [10, 20, 30]
+    conv_prob_fx = copy_prob_forecast_with_axis(
+        prob_forecasts, prob_forecasts.axis, constant_values=const_values)
+    prfxobs = datamodel.ForecastObservation(conv_prob_fx, single_observation)
+
+    fx_values = pd.DataFrame(np.random.randn(10, 3)+10,
+                             index=create_dt_index(10))
+    obs_values = pd.Series(np.random.randn(10)+10,
+                           index=create_dt_index(10))
+    proc_prfx_obs = create_processed_fxobs(prfxobs, fx_values, obs_values)
+    result = calculator.calculate_metrics([proc_prfx_obs], LIST_OF_CATEGORIES,
+                                          probabilistic._REQ_REF_FX)
```

Comment:
I'm confused, I thought this would raise a RuntimeError since no reference is provided

Reply:
Ah it does but is caught in calculate metrics and logs an error instead. Should assert that the logging happened here.

Reply:
I've added logging assertions for both probabilistic and deterministic (had to add new test for that)